java/ql/src/semmle/code/java/frameworks/guava/Collections.qll

```diff
@@ -3,104 +3,534 @@
 import java
 private import semmle.code.java.dataflow.DataFlow
 private import semmle.code.java.dataflow.FlowSteps
+private import semmle.code.java.dataflow.ExternalFlow
 private import semmle.code.java.Collections
 
 private string guavaCollectPackage() { result = "com.google.common.collect" }
 
-/** A reference type that extends a parameterization of one of the various immutable container types. */
-private class ImmutableContainerType extends RefType {
-  string kind;
-
-  ImmutableContainerType() {
-    this.getSourceDeclaration().getASourceSupertype*().hasQualifiedName(guavaCollectPackage(), kind) and
-    kind = ["ImmutableCollection", "ImmutableMap", "ImmutableMultimap", "ImmutableTable"]
-  }
-
-  /**
-   * Gets the name of the most general superclass of this type
-   * from among `ImmutableCollection`, `ImmutableMap`, `ImmutableMultimap`, and `ImmutableTable`.
-   */
-  string getKind() { result = kind }
-}
-
-/** A nested `Builder` class of one of the various immutable container classes */
-private class ContainerBuilder extends NestedType {
-  ContainerBuilder() {
-    this.hasName("Builder") and
-    this.getEnclosingType() instanceof ImmutableContainerType
+private class GuavaCollectCsv extends SummaryModelCsv {
+  override predicate row(string row) {
+    row =
+      [
+        //"package;type;overrides;name;signature;ext;inputspec;outputspec;kind",
+        // Multisets
+        "com.google.common.collect;Multiset;true;add;(Object,int);;Argument[0];Element of Argument[-1];value",
+        "com.google.common.collect;Multiset;true;setCount;(Object,int);;Argument[0];Element of Argument[-1];value",
+        "com.google.common.collect;Multiset;true;setCount;(Object,int,int);;Argument[0];Element of Argument[-1];value",
+        "com.google.common.collect;Multiset;true;elementSet;();;Element of Argument[-1];Element of ReturnValue;value",
+        "com.google.common.collect;Multiset;true;entrySet;();;Element of Argument[-1];Element of Element of ReturnValue;value",
+        "com.google.common.collect;Multiset<>$Entry;true;getElement;();;Element of Argument[-1];ReturnValue;value",
```

Comment:
All of these `<>` qualifiers should no longer be required

Reply:
Could you rebase https://github.com/github/codeql/pull/5796 onto main so I can more easily make this change?

Reply:
Done

Reply:
Hmm; I don't seem to be able to get this change working; removing the `<>`s makes the tests fail. Additionally one of the tests that depends on `Map.Entry` is also failing. It seems like the change that should have made the `<>`s unneeded hasn't actually worked.

Reply:
It depends on the new extractor: make sure you're running codeql 2.5.6

Reply:
Ah, thanks